### PR TITLE
User Auth set up

### PR DESCRIPTION
**Pull request by Star-tide**
I tested all of the endpoints

/api/v1/users/signup/
/api/v1/users/login/ 
/api/v1/users/logout/ 

user is created with...

email, password and display_name

note: I did remove migrations and reapply them

### CODE REVIEW

**Comment by Star-tide** [comment]
Screenshot 2024-08-27 at 4 13 17 PM

**Review by JosephL17 — COMMENTED** [review]
looks good to me

**Comment by Star-tide** [comment]
Make sure you approve!

**Review by Ebasko7 — APPROVED** [review]
Looks good. Approved

**Comment by polly-rado** [comment]
I agree. This looks great. Approved

**Review by polly-rado — APPROVED** [review]
looks great.  Approved

**Comment by Star-tide** [comment]
Will wait for cody, just so he can get a feel

File: backend/user/urls.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from django.urls import path
+from .views import Sign_up, Log_in, Log_out
+
+urlpatterns = [
+    path("signup/", Sign_up.as_view()),
+    path("login/", Log_in.as_view()),
+    path("logout/", Log_out.as_view()),
```
Comment (rck5005) [inline backend/user/urls.py]: Suggest adding "name='usersignup'" or some standardized way to name each url, to identify them for testing.

File: backend/user/views.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            return Response({"User":new_user.display_name, "Email": new_user.email, "Token":token.key, "Success": "User has been successfully created"}, status=HTTP_201_CREATED)
+        except ValidationError as e:
+            print(e)
+            return Response(e, status=HTTP_400_BAD_REQUEST)
+        
+class Log_in(APIView):
+    def post(self, request):
+        data = request.data.copy()
+        user = authenticate(username=data.get("email"), password=data.get("password"))
+        print(user)
```
Comment (rck5005) [inline backend/user/views.py]: For standardization, recommend commenting out debugging lines of code, like this one, before merging.

**Review by rck5005 — APPROVED** [review]
Just threw in some suggestions for standardization. Looks like it'll run clean!

And thanks for giving me the chance to review.

File: backend/user/urls.py (hunk 1 above)
Comment (Star-tide) [reply backend/user/urls.py]: Done

File: backend/user/views.py (hunk 2 above)
Comment (Star-tide) [reply backend/user/views.py]: Done

**Pull request merged**